app/src/test/java/org/simple/clinic/scanid/ScanSimpleIdScreenControllerTest.kt

```diff
@@ -166,6 +168,51 @@ class ScanSimpleIdScreenControllerTest {
     verifyNoMoreInteractions(screen)
   }
 
+  @Test
+  fun `when invalid short code is entered then show validation error`() {
+    //given
+    val shortCodeText = "3456"
+    val shortCodeInput = ShortCodeInput(shortCodeText)
+
+    //when
+    uiEvents.onNext(ShortCodeSearched(shortCodeInput))
+
+    //then
+    verify(screen).showShortCodeValidationError()
+    verifyNoMoreInteractions(screen)
+  }
+
+  @Test
+  fun `when short code text changes, then hide validation error`() {
+    //given
+    val invalidShortCode = "3456"
+    val invalidShortCodeInput = ShortCodeInput(invalidShortCode)
+
+    uiEvents.onNext(ShortCodeSearched(invalidShortCodeInput))
```

Comment:
This is already being tested in the previous test. Is this necessary to add again here?